modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridDhtAtomicCache.java

```diff
@@ -1038,91 +1050,108 @@ private IgniteInternalFuture updateAll0(
         final GridCacheOperation op,
         boolean async
     ) {
-        assert ctx.updatesAllowed();
+        try (MTC.TraceSurroundings ignored =
+                 MTC.support(ctx.kernalContext().tracing().create(CACHE_API_UPDATE, MTC.span()))) {
+            MTC.span().addTagOrLog("cache", CACHE_API_UPDATE, () -> Objects.toString(cacheCfg.getName()));
 
-        ctx.checkSecurity(SecurityPermission.CACHE_PUT);
+            Map<? extends K, ? extends V> m = map;
```

Comment:
map can be declared as final in the argument list. better to log keys count only once:

```
if (map != null)
MTC.span().addTagOrLog("keys.count", CACHE_API_UPDATE,
                () -> String.valueOf(map.size()));

if (invokeMap!= null)
MTC.span().addTagOrLog("keys.count", CACHE_API_UPDATE,
                () -> String.valueOf(invokeMap.size()));
```

Reply:
Fixed.